kernel/Kconfig

```
@@ -382,6 +382,16 @@ config INIT_STACKS
 	  water mark can be easily determined. This applies to the stack areas
 	  for threads, as well as to the interrupt stack.
 
+config NO_BSS_INIT
+	bool "Disable SW BSS initialization"
```

Probably we should make it non-interactive option - so this option can only be selected in SoC/board Kconfig and user can't accidentally enable it by configuration tool (and break their system).

```suggestion
	bool
```

---

Hi @evgeniy-paltsev , I would ask to leave it configurable, since otherwise we wouldn't be able to tweak it through the command line parameter for existing platforms (which is convenient for CI) or through the _defconfig/.cfg (for e.g. RTL-simulator oriented applications.

---

No, I agree with @evgeniy-paltsev. This should not be user-configurable, it should be determined by the platform. If you want to tweak it at the board level you can use the board's `Kconfig.defconfig` file

---

All right, let's change it. 
We still can keep it configurable in our local tree, so no big deal.

Thanks!